File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        int[] lastIdx = IntStream.generate(() -> -1).limit(26).toArray();

        for (int i = 0; i < s.length(); i++) {
            int idx = s.charAt(i) - 'a';
            int last = lastIdx[idx];
            answer[i] = (last == -1) ? -1 : (i - last);
            lastIdx[idx] = i;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = [-1] * len(s)
    last_idx = { chr(c) : -1 for c in range(ord('a'), ord('z') + 1) }

    for i, ch in enumerate(s):
        last = last_idx[ch]
        answer[i] = -1 if last == -1 else i - last
        last_idx[ch] = i

    return answer